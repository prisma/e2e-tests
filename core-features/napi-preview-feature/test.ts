import { PrismaClient } from '@prisma/client'
import fs from 'fs'

describe('N-API', () => {
  test('has n-api library', async () => {
    const files = fs.readdirSync('./node_modules/.prisma/client')
    expect(files).toMatchSnapshot()
  })
  test('generated client', async () => {
    const prisma = new PrismaClient()
    await prisma.$connect()
    await prisma.user.deleteMany()
    const users = await prisma.user.findMany()
    expect(users).toMatchSnapshot()
    const usr = await prisma.user.create({
      data: {
        id: 1,
        name: 'Test',
        email: 'test@example.com',
        posts: {
          create: {
            id: 1,
            title: 'Test',
          },
        },
      },
      include: {
        posts: true,
      },
    })
    const post = await prisma.post.findUnique({
      where: {
        id: usr.posts[0].id,
      },
    })
    expect(post).toMatchSnapshot()
    const postsDeleted = await prisma.post.deleteMany()
    const usersDeleted = await prisma.user.deleteMany()
    expect({
      users: usersDeleted,
      posts: postsDeleted,
    }).toMatchSnapshot()
  })
})
