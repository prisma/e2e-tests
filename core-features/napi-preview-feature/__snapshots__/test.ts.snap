// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`N-API Off 1`] = `
Object {
  "delete": Object {
    "posts": Object {
      "count": 1,
    },
    "users": Object {
      "count": 1,
    },
  },
  "post.findUnique": Object {
    "id": 1,
    "title": "Test",
    "userId": 1,
  },
  "user.create": Object {
    "email": "test@example.com",
    "id": 1,
    "name": "Test",
    "posts": Array [
      Object {
        "id": 1,
        "title": "Test",
        "userId": 1,
      },
    ],
  },
  "users.findMany": Array [],
}
`;

exports[`N-API Off 2`] = `
"yarn run v1.22.10
$ /home/will/Prisma/e2e-tests/core-features/napi-preview-feature/node_modules/.bin/prisma -v
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
Query Engine         : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
Done in 0.90s."
`;

exports[`N-API Off: ./node_modules/.prisma/client 1`] = `
Array [
  "index-browser.js",
  "index.d.ts",
  "index.js",
  "package.json",
  "query-engine-debian-openssl-1.1.x",
  "schema.prisma",
]
`;

exports[`N-API Off: ./node_modules/@prisma/engines 1`] = `
Array [
  "README.md",
  "dist",
  "download",
  "introspection-engine-debian-openssl-1.1.x",
  "migration-engine-debian-openssl-1.1.x",
  "package.json",
  "prisma-fmt-debian-openssl-1.1.x",
  "query-engine-debian-openssl-1.1.x",
]
`;

exports[`N-API Off: ./node_modules/prisma 1`] = `
Array [
  "README.md",
  "build",
  "install",
  "package.json",
  "preinstall",
  "prisma-client",
  "query-engine-debian-openssl-1.1.x",
  "scripts",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true 1`] = `
Object {
  "delete": Object {
    "posts": Object {
      "count": 1,
    },
    "users": Object {
      "count": 1,
    },
  },
  "post.findUnique": Object {
    "id": 1,
    "title": "Test",
    "userId": 1,
  },
  "user.create": Object {
    "email": "test@example.com",
    "id": 1,
    "name": "Test",
    "posts": Array [
      Object {
        "id": 1,
        "title": "Test",
        "userId": 1,
      },
    ],
  },
  "users.findMany": Array [],
}
`;

exports[`N-API PRISMA_FORCE_NAPI=true 2`] = `
"yarn run v1.22.10
$ /home/will/Prisma/e2e-tests/core-features/napi-preview-feature/node_modules/.bin/prisma -v
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
N-API Query Engine   : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
Done in 0.87s."
`;

exports[`N-API PRISMA_FORCE_NAPI=true and Preview Feature 1`] = `
Object {
  "delete": Object {
    "posts": Object {
      "count": 1,
    },
    "users": Object {
      "count": 1,
    },
  },
  "post.findUnique": Object {
    "id": 1,
    "title": "Test",
    "userId": 1,
  },
  "user.create": Object {
    "email": "test@example.com",
    "id": 1,
    "name": "Test",
    "posts": Array [
      Object {
        "id": 1,
        "title": "Test",
        "userId": 1,
      },
    ],
  },
  "users.findMany": Array [],
}
`;

exports[`N-API PRISMA_FORCE_NAPI=true and Preview Feature 2`] = `
"yarn run v1.22.10
$ /home/will/Prisma/e2e-tests/core-features/napi-preview-feature/node_modules/.bin/prisma -v
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
N-API Query Engine   : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
Preview Features     : placeholder
Done in 0.97s."
`;

exports[`N-API PRISMA_FORCE_NAPI=true and Preview Feature: ./node_modules/.prisma/client 1`] = `
Array [
  "index-browser.js",
  "index.d.ts",
  "index.js",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "schema.prisma",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true and Preview Feature: ./node_modules/@prisma/engines 1`] = `
Array [
  "README.md",
  "dist",
  "download",
  "introspection-engine-debian-openssl-1.1.x",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "migration-engine-debian-openssl-1.1.x",
  "package.json",
  "prisma-fmt-debian-openssl-1.1.x",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true and Preview Feature: ./node_modules/prisma 1`] = `
Array [
  "README.md",
  "build",
  "install",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "preinstall",
  "prisma-client",
  "scripts",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true: ./node_modules/.prisma/client 1`] = `
Array [
  "index-browser.js",
  "index.d.ts",
  "index.js",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "schema.prisma",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true: ./node_modules/@prisma/engines 1`] = `
Array [
  "README.md",
  "dist",
  "download",
  "introspection-engine-debian-openssl-1.1.x",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "migration-engine-debian-openssl-1.1.x",
  "package.json",
  "prisma-fmt-debian-openssl-1.1.x",
]
`;

exports[`N-API PRISMA_FORCE_NAPI=true: ./node_modules/prisma 1`] = `
Array [
  "README.md",
  "build",
  "install",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "preinstall",
  "prisma-client",
  "scripts",
]
`;

exports[`N-API Preview Feature 1`] = `
Object {
  "delete": Object {
    "posts": Object {
      "count": 1,
    },
    "users": Object {
      "count": 1,
    },
  },
  "post.findUnique": Object {
    "id": 1,
    "title": "Test",
    "userId": 1,
  },
  "user.create": Object {
    "email": "test@example.com",
    "id": 1,
    "name": "Test",
    "posts": Array [
      Object {
        "id": 1,
        "title": "Test",
        "userId": 1,
      },
    ],
  },
  "users.findMany": Array [],
}
`;

exports[`N-API Preview Feature 2`] = `
"yarn run v1.22.10
$ /home/will/Prisma/e2e-tests/core-features/napi-preview-feature/node_modules/.bin/prisma -v
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
Query Engine         : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
Preview Features     : placeholder
Done in 0.92s."
`;

exports[`N-API Preview Feature: ./node_modules/.prisma/client 1`] = `
Array [
  "index-browser.js",
  "index.d.ts",
  "index.js",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "schema.prisma",
]
`;

exports[`N-API Preview Feature: ./node_modules/@prisma/engines 1`] = `
Array [
  "README.md",
  "dist",
  "download",
  "introspection-engine-debian-openssl-1.1.x",
  "migration-engine-debian-openssl-1.1.x",
  "package.json",
  "prisma-fmt-debian-openssl-1.1.x",
  "query-engine-debian-openssl-1.1.x",
]
`;

exports[`N-API Preview Feature: ./node_modules/prisma 1`] = `
Array [
  "README.md",
  "build",
  "install",
  "libquery_engine_napi-debian-openssl-1.1.x.so.node",
  "package.json",
  "preinstall",
  "prisma-client",
  "query-engine-debian-openssl-1.1.x",
  "scripts",
]
`;
